src/GraphQL/Validation/TypeInfo.cs

```diff
@@ -34,7 +32,14 @@ public TypeInfo(ISchema schema)
         /// <returns></returns>
         public INode[] GetAncestors()
         {
-            return _ancestorStack.Skip(1).Reverse().ToArray();
+            var ancestorArray = _ancestorStack.ToArray();
+            var c = ancestorArray.Length;
+            var ret = new INode[--c];
+            for (int i = c; i > 0; --i) {
+                ret[c - i] = ancestorArray[i];
+            }
+            return ret;
+            // aka: return _ancestorStack.Skip(1).Reverse().ToArray();
```

Comment:
I am taking over the modification of this method myself in the next PR.

Reply:
Fine with me.  I would be interested to see the performance comparison, as it's already down to only two allocations.

Reply:
Oh I forgot - you said you were going to use an index.  So it should have only one allocation.